### PR TITLE
On version numbering, from discussion in PR #34535

### PR DESCRIPTION
**Pull request by marbx**
### What does this PR do?
Documenting the relationship between version numbering and Git 

From discussion in PR #34535

### CODE REVIEW

File: doc/topics/releases/version_numbers.rst [hunk 1]
```diff
@@ -46,3 +46,50 @@ to be cut and the month is ``June``. This would make the new release number
 
 After the release is cut, new features would be worked on under the ``Xenon``
 code name and the process repeats itself.
+
+
+Version numberss, Git and salt --version
```
Comment (cachedout) [inline doc/topics/releases/version_numbers.rst]: `Version numbers`

File: doc/topics/releases/version_numbers.rst [hunk 2]
```diff
@@ -46,3 +46,50 @@ to be cut and the month is ``June``. This would make the new release number
...
+-------
+The salt version, for prorgammers, is based on `git describe` and presented to end users with `salt --version`.
```
Comment (cachedout) [inline doc/topics/releases/version_numbers.rst]: `programmers`

Comment (cachedout) [inline doc/topics/releases/version_numbers.rst]: `end-users`

**Review by cachedout — CHANGES_REQUESTED** [review]

**Comment by markuskramerIgitt** [comment]
Hello @cachedout, thank you for your corrections.

I fxied my spelling errors and checked in 33c1978, which appears above.
Does this mean I don't need another Pull Request?

File: doc/topics/releases/version_numbers.rst [hunk 3]
```diff
@@ -46,3 +46,50 @@ to be cut and the month is ``June``. This would make the new release number
...
+
+Arguments for `git checkout`
+  +------------+----------------------------------------------------------------------------+
+  |  Argument  |                                           Comment                          |
+  +============+============================================================================+
+  | v2016.11   | Tag of the moment that the 2016.11 branch was created from develop         |
+  +------------+----------------------------------------------------------------------------+
+  | 2016.11    | **Release branch** Actively developed bug-fixes for all 2016.11.* releases |
+  +------------+----------------------------------------------------------------------------+
+  | v2016.11.1 | Tag of the moment that the 2016.11.1 branch was created                    |
```
Comment (rallytime) [inline doc/topics/releases/version_numbers.rst]: Change the word `branch` to `release` here. The `v2016.11.1` syntax is not a branch. It is a tag signaling the commit that a release is based on.

**Review by rallytime — CHANGES_REQUESTED** [review]
I had one small technicality about the version_numbers doc changes. Once that is fixed, I think the doc changes look good. Thanks for putting this together @markuskramerIgitt!

However, I did not review the other files included here, as the windows build files are not my area. Were those files meant to be included here?

File: doc/topics/releases/version_numbers.rst (hunk 3 above)
Comment (markuskramerIgitt) [inline doc/topics/releases/version_numbers.rst]: @rallytime, thank you, I changed the wording (on my local file).

In fact there are many unrelated files in this pull request. This was not my intention.
I will now make a local backup of my pending changes, cancel all my pull requests, delete ally my branches and start over from scratch.

**Pull request closed without merging**